packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart

```diff
@@ -367,6 +371,7 @@ abstract class RenderSliverMultiBoxAdaptor extends RenderSliver
       childParentData._keptAlive = true;
     } else {
       assert(child.parent == this);
+      indexes.add(childParentData.index);
```

Comment:
I'm confused here why `removeChild` below doesn't do what we want here.  Calling `collectGarbage` later is just going to invoke `removeChild` again right?

Reply:
Ah, I see now from your other example.